src/main/java/org/jvnet/solaris/libzfs/ZFSObject.java

```diff
@@ -503,10 +506,54 @@ public int callback(zfs_handle_t handle, Pointer arg) {
     }
 
     /**
+     * Grants the specified set of permissions to this dataset.
+     * NOTE that the "zfs_perm_*" libzfs.so routines are not present at
+     * all in OpenZFS, nor even in Sun Solaris 10u8 (last seen in 10u6).
+     * Not sure about Sol10u7 since it was a rather intermediate release.
+     * They may have been replaced with other logic, since zfs_allow_t and
+     * related data types and comments are still present in the headers.
+     */
+    public void allow(ACLBuilder acl) {
```

Comment:
Apparently, there are new routines like `zfs_get_fsacl` and `zfs_set_fsacl` at least in OpenZFS (but not present in Sol10u8):

https://github.com/illumos/illumos-gate/blob/5c653870ec5357c65dc7d975a8c34619d9da0566/usr/src/cmd/zfs/zfs_main.c#L5375